### ISSUE TITLE
Commented out CSS string breaks syntax highlighting for rest of file

### ISSUE THREAD

**Issue by MorningLightMountain713** [Textualize/tcss-vscode-extension#6]
Eg

```python
    class Test(Widget):
        # DEFAULT_CSS = """
        #     Test {
        #         width: auto;
        #         height: auto;
        #     }
        # """
```

Makes the rest of the file change colors. The first triple quote is still visible.

Here is a screenshot:

<img width="358" alt="Screenshot 2023-11-04 at 8 14 58 AM" src="https://github.com/Textualize/tcss-vscode-extension/assets/22514713/e128346a-6036-4ee5-a977-d1a57dfbac71">

**Comment by rodrigogiraoserrao** [comment]
Hey @MorningLightMountain713, thanks for reporting this issue!

As a temporary workaround, may I suggest that you comment out the `DEFAULT_CSS` with triple single-quotes?
Like so:

```py
class Test(Widget):
    '''
    DEFAULT_CSS = """
        Test {
            width: auto;
            height: auto;
        }
    """
    '''
```

This won't break highlighting for the remainder of the file.

**Comment by willmcgugan** [comment]
@rodrigogiraoserrao What would be involved in fixing this?

**Comment by rodrigogiraoserrao** [comment]
> @rodrigogiraoserrao What would be involved in fixing this?

A “quick & dirty” solution:
right now, we detect TCSS inside Python by using a regex that looks for things like `DEFAULT_CSS = """` and the matching `"""`.
I could change the regex to make sure that the only thing coming before the class variable is whitespace.

Pathological cases like the one below would still fail, but maybe they're so pathological that it is ok:

```py
class Test(Widget):
    some_class_var = 3; DEFAULT_CSS = """Test { width: auto; height; auto }"""
```

For the more foolproof solution I'd need to check the VS Code extension docs.

**Comment by willmcgugan** [comment]
I see. I assumed it was more clever than that. So it would break if you used single quotes?

If regex is the only solution then I think it will be difficult if not impossible to completely guard against it.

**Comment by rodrigogiraoserrao** [comment]
No, it doesn't break with single quotes and I think the quick & dirty fix is well worth it.

The whole grammar is a bunch of regular expressions because that's just the format for syntax highlighting, there's not much I can do about it.
You can see [here](https://github.com/Textualize/tcss-vscode-extension/blob/56d29749f694f8e2cbdf5ae8e1f60b99a3529d0d/syntaxes/python.injection.json#L6), [here](https://github.com/Textualize/tcss-vscode-extension/blob/56d29749f694f8e2cbdf5ae8e1f60b99a3529d0d/syntaxes/python.injection.json#L19), [here](https://github.com/Textualize/tcss-vscode-extension/blob/56d29749f694f8e2cbdf5ae8e1f60b99a3529d0d/syntaxes/python.injection.json#L32), and [here](https://github.com/Textualize/tcss-vscode-extension/blob/56d29749f694f8e2cbdf5ae8e1f60b99a3529d0d/syntaxes/python.injection.json#L45), the four regular expressions that find TCSS embedded in Python.
They can be improved a bit, but like you say they'll hardly ever be perfect.

For smarter things, I think we'd need a language server that runs in the background.
If we end up doing things like linting action names + action methods, message handler methods, watcher methods, etc, you may end up having a language server in the future either way.
As of now, I have no idea how difficult it is to have a basic one running, but I've seen docs pages about it.

**Comment by willmcgugan** [comment]
Given there is a workaround, suggest we park this for now. We can revisit it in the future to make it a bit smarter.
